global-news-app/src/Charts/ChartsSection/PihChart/PihChart.tsx

```diff
@@ -0,0 +1,81 @@
+import ChartCard from "../ChartCard/ChartCard";
+import EmptyState from "components/EmptyState/EmptyState";
+import { ChartsProps } from "../ChartsSection";
+import { PihChartContainer } from "./PihChart.style";
+import { PieChart, Pie, Cell } from "recharts";
+import { Charts } from "Types/enums";
+import Legend from "./Legend/Legend";
+import { COLORS } from "Utils/consts";
+import { isEmpty } from "lodash";
+
+interface ISourceCounts {
+  [name: string]: number;
+}
+export interface ISource {
+  name: string;
+  value: number;
+  color?: string;
+}
+
+const PihChart = ({ articles }: ChartsProps) => {
+  const formattedData = () => {
+    const sourceCounts: ISourceCounts = {};
+
+    articles?.forEach((article) => {
+      const sourceName = article.source.name;
+      sourceCounts[sourceName] = (sourceCounts[sourceName] || 0) + 1;
+    });
+
+    const sourceArray: ISource[] = Object.keys(sourceCounts).map((name) => ({
+      name,
+      value: sourceCounts[name],
+    }));
+    return sourceArray;
+  };
+
+  const data = formattedData();
+
+  return (
+    <ChartCard type={Charts.Sources}>
+      {isEmpty(data) ? (
+        <EmptyState marginTop="38" />
+      ) : (
+        <PihChartContainer>
+          <PieChart width={124} height={124}>
+            <Pie
+              data={data}
+              dataKey="value"
+              nameKey="name"
+              cx="50%"
+              cy="50%"
+              outerRadius={62}
+              innerRadius={54}
+              stroke=""
+            >
+              {data?.map((_, index) => (
+                <Cell
+                  key={`cell-${index}`}
+                  fill={COLORS[index % COLORS.length]}
+                />
+              ))}
+            </Pie>
+            <text
+              x={62}
+              y={62}
+              fontFamily="Mulish, sans-serif"
+              color="#030035"
+              fontSize={12}
+              textAnchor="middle"
+              dominantBaseline="middle"
+            >
+              {Charts.Sum}
```

Comment:
a little hint, this actually should be a number representing a sum, and not the string "Sum"